mne/time_frequency/spectrum.py

```diff
@@ -352,9 +346,12 @@ def __init__(
         )
         if method_kw.get("average", "") in (None, False):
             self._dims += ("segment",)
+        if self._returns_complex_tapers(**method_kw):
+            self._dims = self._dims[:-1] + ("taper",) + self._dims[-1:]
         # record data type (for repr and html_repr)
         self._data_type = (
-            "Fourier Coefficients" if "taper" in self._dims else "Power Spectrum"
+            f"{'Complex' if method_kw.get('output', '') == 'complex' else 'Real'} "
+            "Spectrum"
         )
```

Comment:
This was something discussed in #11803, not sure how people feel about it now.

If people prefer the current distinction of `"Fourier coeffs"` vs. `"Power Spectrum"`, I think a change would need to be made anyway since Welch method does not have taper dimension but could be complex-valued (taper only applies to complex multitaper output).

Reply:
Got it, then this keeps the same repr but just adapts the criteria to also work for Welch.

Reply:
I see your point. Then this retains the repr but makes sure the logic works for Welch as well:
https://github.com/mne-tools/mne-python/blob/361db716396adfbe4ea0a2f9b837bc7ccd132dd4/mne/time_frequency/spectrum.py#L352-L356